src/hotspot/share/gc/shared/gcTraceTime.cpp

```diff
@@ -70,32 +71,39 @@ void GCTraceTimeLoggerImpl::log_end(Ticks end) {
   out.print_cr(" %.3fms", duration_in_ms);
 }
 
-GCTraceCPUTime::GCTraceCPUTime() :
-  _active(log_is_enabled(Info, gc, cpu)),
+GCTraceCPUTime::GCTraceCPUTime(GCTracer* tracer) :
+  _active(log_is_enabled(Info, gc, cpu) ||
+          (tracer != nullptr && tracer->should_report_cpu_time_event())),
   _starting_user_time(0.0),
   _starting_system_time(0.0),
-  _starting_real_time(0.0)
+  _starting_real_time(0.0),
+  _tracer(tracer)
 {
   if (_active) {
     bool valid = os::getTimesSecs(&_starting_real_time,
-                               &_starting_user_time,
-                               &_starting_system_time);
+                                  &_starting_user_time,
+                                  &_starting_system_time);
     if (!valid) {
       log_warning(gc, cpu)("TraceCPUTime: os::getTimesSecs() returned invalid result");
       _active = false;
     }
   }
 }
 
+GCTraceCPUTime::GCTraceCPUTime() { GCTraceCPUTime(nullptr); }
```

Comment:
```suggestion
GCTraceCPUTime::GCTraceCPUTime() : GCTraceCPUTime(nullptr) { }
```
The way the other constructor is called is wrong afaik. However if serial gc support were added, this constructor could be deleted.

Reply:
Done.